contracts/hub/virtual_balance/src/tests.rs

```diff
@@ -145,4 +146,105 @@ mod tests {
         assert_eq!(expected_snapshot_balance_user_1, snapshot_balance);
         assert_eq!(expected_snapshot_balance_user_2, snapshot_balance_2);
     }
+
+    #[test]
```

Comment:
Can we add a test case where we expect some things to fail? Maybe a case where Approval isn't granted.